noir-projects/noir-protocol-circuits/crates/public-kernel-lib/src/public_kernel_teardown.nr

```diff
@@ -22,6 +22,45 @@ impl PublicKernelTeardownCircuitPrivateInputs {
         assert(needs_teardown == true, "Cannot run unnecessary teardown circuit");
     }
 
+    // Validates that the start gas injected into the app circuit matches the teardown gas limits set by the user
+    fn validate_start_gas(self) {
+        let public_call_start_gas = self.public_call.call_stack_item.public_inputs.start_gas_left;
+        let teardown_gas_limit = self.previous_kernel.public_inputs.constants.tx_context.gas_settings.teardown_gas_limits;
+        assert(
+            public_call_start_gas == teardown_gas_limit, "Start gas for teardown phase does not match teardown gas allocation"
+        );
+    }
+
+    // Validates the transaction fee injected into the app circuit is properly computed from gas_used and block gas_fees
+    fn validate_transaction_fee(self) {
+        let transaction_fee = self.public_call.call_stack_item.public_inputs.transaction_fee;
+        // Note that teardown_gas is already included in end.gas_used as it was injected by the private kernel
+        let total_gas_used = self.previous_kernel.public_inputs.end.gas_used.add(self.previous_kernel.public_inputs.end_non_revertible.gas_used);
+        // TODO(palla/gas): Load gas fees from a PublicConstantData struct that's currently missing
```

Comment:
If we decide to put the current prices in the previous block, we can do this now.

Reply:
I'm still not sure about that, but we can do it as a stopgap. I'll do on a subsequent PR!

Reply:
Here it goes https://github.com/AztecProtocol/aztec-packages/pull/6010!